docs/ocis/adr/0022-sharing-and-space-management-api.md

```diff
@@ -0,0 +1,81 @@
+---
+title: "22. Sharing and Space Management API"
+---
+
+* Status: accepted
+* Deciders: @jammingben @butonic @theonering @kobergj @micbar
+* Date: 2023-08-08
+
+Technical Story: [Public issue](https://github.com/owncloud/ocis/issues/6993)
+
+## Context and Problem Statement
+
+In the early days of the rewrite of ownCloud it was an important goal to keep all important APIs compatible with ownCloud 10. Infinite Scale embraced that goal until version 1.0.0.
+
+After that first release, the focus changed.
+
+Infinite Scale started the spaces feature which brings a whole new set of APIs and concepts. We made the conscious decision to keep the sharing API as it was, live with its shortcomings and create workarounds to support spaces. We have come a long way so far. Now we need to move on. The Web Client has made the decision to drop the support of ownCloud 10 and keep version 7.0.0 alive for ownCloud 10 to keep the easy migration path intact.
+
+The desktop and mobile client platforms were suffering from poor support from the server and can now move forward with a new API implementation. By using openApi 3 and all the needed tooling around it developing the LibreGraph specification, documentaion and SDKs, we now feel confident to move on.
+
+## Decision Drivers
+
+* The Path based nature of the OCS API lacks spaces support
+* The permissions bitmask is no longer working when using sharing roles
+* We want to support server announced sharing roles which are different per instance or scope
+* We need to get rid of the currently hardcoded sharing roles in our clients
+* New sharing roles and permissions are needed to support secure view and other new features
```

Comment:
slightly off topic: the new permissions (secure view) will still be available via webdav?

Reply:
This needs to be discussed in detail in the api spec.

I was thinking of an extra api call "what can i do on this resource?". The webdav permissions are not covering all the fine grained permissions and are not reporting which sharing roles will be available on this resource.

Let me give an example:

We have a space root, 3 roles are available "Manager" "Editor" "Viewer". The space root is shared with Marie with "Viewer".
Marie goes to a subfolder "Space-Root-A/Sub/Folder" and wants to change the permissions on the subfolder. The answer from the api would be: "Editor" "Deny" because these are the only possible options on that subfolder.

In "Space-Root-B/Sub/Folder" the deny feature would be switched off, and the response  from the api would be: "Editor".

The client can do a "What can i do on this resource with this resourceID" request and the answer would be

Reply:
@butonic ^

Reply:
I'd really like to see sharing information available inline with the WebDAV `PROPFIND` response, through a new `oc:` tag.

The contents of that tag can then carry JSON describing permissions/roles/… in detail - using the same format as new endpoints for creating/modifying/listing shares/links would use.

Reply:
mixing webdav and json 😱

Reply:
Well, that'd be the price to pay to be able to use the same structures in both PROPFIND and new endpoints. 🤷‍♂️

Unless, of course, both should use XML. 🥴

Reply:
@felix-schwarz Let us not blow up the WebDAV further and further. It was not designed for that purpose.
I consider WebDAV as the legacy File Authoring Protocol because there is nothing better.

The shares were also reported on the OCS endpoint before, so now they move from OCS to libregraph.